tools/scripts/generateSharedConstants.rb

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env ruby
+#
+# This file generates a static json file, and uses it to create enum-like objects
+# that can be used in both dashboard and apps. It does this by generating two
+# static files - one which is a ruby file to be used by dashboard, the other
+# which is a js file to be used by apps. In both cases, the expectation is that
+# the generated files get checked into the repo
+#
```

Comment:
Open question. My next commit (which I'm currently battling rubocop on) adds a rake task, which also gets run as part of rake:build. But if you make changes to the config file, you'll still need to do another build (or rerun the rake task).